storage/rocksdb/ha_rocksdb.cc

```diff
@@ -7102,6 +7117,30 @@ static void rocksdb_truncation_table_cleanup(void) {
   }
 }
 
+static void move_wals_to_target_dir() {
+  if (is_wal_dir_separate()) {
+    if ((my_mkdir(rocksdb_wal_dir, S_IRWXU, MYF(0)) == -1) &&
+        (my_errno() != EEXIST))
+      rdb_fatal_error("Failed to create %s", rocksdb_wal_dir);
+
+    for_each_in_dir(
+        rocksdb_datadir, MY_WANT_STAT | MY_FAE, [](const fileinfo &f_info) {
+          if (!S_ISREG(f_info.mystat->st_mode) ||
+              !has_file_extension(f_info.name, ".log"))
+            return true;
+          LogPluginErrMsg(INFORMATION_LEVEL, ER_LOG_PRINTF_MSG,
+                          "Moving %s to %s", f_info.name, rocksdb_wal_dir);
+          const auto src_path = rdb_concat_paths(rocksdb_datadir, f_info.name);
+          const auto dst_path = rdb_concat_paths(rocksdb_wal_dir, f_info.name);
+          if (my_rename(src_path.c_str(), dst_path.c_str(),
+                        MYF(MY_WME | MY_FAE)) == -1)
+            rdb_fatal_error("Failed to move %s to %s", f_info.name,
```

Comment:
I saw failing MTR test in internal diff tool. It's complaining about the OS errno 18 - Invalid cross-device link.
error message: Error on rename of './.rocksdb/000874.log' to '/tmp/b1Pt6ABQZ_/2/rdb_wal/000874.log' (OS errno 18 - Invalid cross-device link)

I remember we hit the same issue in fixup_rename and fixed that by converting to copy and delete. Looks like we hit the same issue here. Interestingly, I never hit the issue locally and the MTR test seems to pass if I run it locally.

Reply:
rocksdb.wal_dir is still failing in this area. It seems like detection of when to copy&delete vs. when to move/rename might not be robust enough?

Reply:
The test failure happens in move_wals_to_target_dir(). The current detection of when to copy&delete is only in the fixup_on_startup(). We don't have the detection in move_wals_to_target_dir(), which is called after fixup_on_startup(). I think we need to add the detection here as well.

Reply:
Yes, that looks to be the case. `move_wals_to_target_dir()` needs to perform move vs. copy&delete detection.

Reply:
Alternatively, we can always copy&delete WAL. The size of the WALs is small compared to the bulk of the data files, so this is probably acceptable.

Reply:
All the infrastructure to support move over `EXDEV` is already here, so I pulled it up from clone to `myrocks::rdb_` namespace/prefix and called it at this location.

Strictly speaking a part of this patch should be PR'ed separetely and may go in before the clone itself, let me know if you want me to do this.